schneiderWallbox/schneiderwallbox.cpp

```diff
@@ -0,0 +1,245 @@
+                sendCommand = true;
+            }
+        }
+    }
+
+    if (sendCommand) {
+        qCDebug(dcSchneiderElectric()) << "Sending remote command:" << remoteCommandSend;
+        QModbusReply *reply = m_modbusTcpConnection->setRemoteCommand(remoteCommandSend);
+        if (!reply) {
+            qCWarning(dcSchneiderElectric()) << "Sending remote command failed because the reply could not be created.";
+            m_errorOccured = true;
+            return;
+        }
+        m_lastCommand = remoteCommandSend;
+        m_acknowledgeCommand = true;
+        connect(reply, &QModbusReply::finished, reply, &QModbusReply::deleteLater);
+        connect(reply, &QModbusReply::errorOccurred, this, [reply] (QModbusDevice::Error error){
+            qCWarning(dcSchneiderElectric()) << "Modbus reply error occurred while sending remote command" << error << reply->errorString();
+            emit reply->finished(); // To make sure it will be deleted
+        });
+    }
+
+    if (m_chargeCurrent != chargeCurrentSetpointSend) {
+        qCDebug(dcSchneiderElectric()) << "Device max current:" << m_chargeCurrent <<", sending modbus command current: " << chargeCurrentSetpointSend;
+        QModbusReply *reply = m_modbusTcpConnection->setMaxIntensitySocket(chargeCurrentSetpointSend);
+        if (!reply) {
+            qCWarning(dcSchneiderElectric()) << "Sending max intensity socket failed because the reply could not be created.";
+            m_errorOccured = true;
+            return;
+        }
+        connect(reply, &QModbusReply::finished, reply, &QModbusReply::deleteLater);
+        connect(reply, &QModbusReply::errorOccurred, this, [reply] (QModbusDevice::Error error){
+            qCWarning(dcSchneiderElectric()) << "Modbus reply error occurred while sending max intensity socket" << error << reply->errorString();
+            emit reply->finished(); // To make sure it will be deleted
+        });
+    }
+}
+
+bool SchneiderWallbox::enableOutput(bool state)
+{
+    if (!m_modbusTcpConnection->connected()) {
+        qCDebug(dcSchneiderElectric()) << "Can't set charging on/off, device is not connected.";
+        return false;
+    }
+
+    qCDebug(dcSchneiderElectric()) << "Setting wallbox output to" << state;
+
+    m_charging = state;
+    //update();  Don't do update here, because the wallbox does not react very fast to modbus writes.
+    if (m_errorOccured) {
+        return false;
+    }
+    return true;
+}
+
+bool SchneiderWallbox::setMaxAmpere(int ampereValue)
+{
+    if (!m_modbusTcpConnection->connected()) {
+        qCDebug(dcSchneiderElectric()) << "Can't set current limit, device is not connected.";
+        return false;
+    }
+
+    qCDebug(dcSchneiderElectric()) << "Setting wallbox max current to" << ampereValue;
+
+    m_chargeCurrentSetpoint = ampereValue;
+    //update();  Don't do update here, because the wallbox does not react very fast to modbus writes.
+    if (m_errorOccured) {
+        return false;
```

Comment:
Hmm.. I can see how you got there. The first intention was probably to directly write the register but then it turned out the whole update() is too slow and blabla... Can't tell you what the way to go is (perhaps what you do now is the best thing to do) but what strikes me here is that you seem to evaluate an outdated m_errorOccurred... Perhaps even that is ok (I can't test it without having such a wallbox) but please take a minute to think about what exactly happens and the corner cases it could run into the effects of return false here are (afaict it will at least cause the ThingActionInfo to finish() with an error, even though it might actually succeed...